site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+            },
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    const context: AppContext = call.context;
+
+    // Extract the user's team from the expanded context
+    const team: Team = context.team;
+
+    // Create a permalink to put at the bottom of the ticket description
+    const permalink = `${context.mattermost_site_url}/${team.name}/pl/${context.post_id}`;
+    const ticketFooter = `_Created from a [post](${permalink}) in Mattermost_`
+
+    const fullDescription = `${description}\n\n${ticketFooter}`;
+
+    // Some ticket from another system, e.g. Jira or ZenDesk
+    let ticket: Ticket;
+    try {
+        ticket = await createTicket(summary, fullDescription);
+    } catch (e) {
+        // The modal will remain open on the user's screen, and this error will be shown at the bottom of the form.
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Error creating the ticket: ${e.message}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    // Extract the acting user from the expanded context so we can access their username.
+    const actingUser: UserProfile = context.acting_user;
+
+    // Create a public post using the bot token. This requires the `act_as_bot` permission to be in your App manifest.
+    const publicMessage = `@${actingUser.username} created ticket [${ticket.id}: ${summary}](${ticket.url})`;
+    await createPostAsBot(call.context, publicMessage);
+
+    const privateResponseToUser = 'Thanks for making the ticket! Feel free to create more :)';
+    const callResponse: AppCallResponse = {
+        type: 'ok',
+        markdown: privateResponseToUser,
+    }
+
+    res.json(callResponse);
+});
+
+const createPostAsBot = async (context: AppContext, message: string): Promise<Post> => {
+    const botClient = new Client4();
+    botClient.setUrl(context.mattermost_site_url);
+    botClient.setToken(context.bot_access_token);
+
+    const rootID = context.root_post_id || context.post_id;
+
+    const post = {
+        message,
+        channel_id: context,
+        root_id: rootID,
+    } as Post;
+
+    return botClient.createPost(post);
+};
+```
+
+Now we have a functioning form that allows the user to attach a post to a new ticket, along with a reply `Post` made by our bot that links to the new ticket.
+
+### Static select values
+
+This form works for simple use cases, but ticket systems are usually more complicated than this. What if we have some labels the user needs to choose from? We'll call the labels "tags" in this example so as to not get confused with the `label` field for the select options.
+
+If the App needs to provide a fixed set of values, we can use a `static_select` field. Let's add this field to our form's `fields` array:
+
+```ts
+const tagsField: AppField = {
+    name: 'tags',
+    modal_label: 'Tags',
+    type: 'static_select',
+    value: [],
```

Comment:
No need to set value, if I am not mistaken.

Reply:
Either way, a cool example for the default value (if we were to want that, we may not) would be to add a default tag `{label: 'Mattermost', value: 'mattermost'}` to identify the ticket as a ticket created from Mattermost.